extensions/sql-database-projects/src/models/project.ts

```diff
@@ -0,0 +1,91 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as xml2js from 'xml2js';
+import * as path from 'path';
+import { promises as fs } from 'fs';
+import { DataSource } from './dataSources/dataSources';
+
+/**
+ * Class representing a Project, and providing functions for operating on it
+ */
+export class Project {
+	public projectFile: string;
+	public files: ProjectEntry[] = [];
+	public dataSources: DataSource[] = [];
+
+	constructor(projectFilePath: string) {
+		this.projectFile = projectFilePath;
+	}
+
+	/**
+	 * Reads the project setting and contents from the file
+	 */
+	public async readProjFile() {
+		let projFileContents = await fs.readFile(this.projectFile);
+
+		const parser = new xml2js.Parser({
+			explicitArray: true,
+			explicitCharkey: false,
+			explicitRoot: false
+		});
+
+		let result;
+
+		try {
+			result = await parser.parseStringPromise(projFileContents.toString());
+		}
+		catch (err) {
+			vscode.window.showErrorMessage(err);
+			return;
+		}
+
+		// find all folders and files to include
+
+		for (const itemGroup of result['ItemGroup']) {
```

Comment:
are these typed? (`result` and `itemGroup`) if not you should consider adding your own typing on them. It helps when other people who aren't familiar with this code want to make changes here that they are just accessing properties that aren't actually there.

Reply:
Hey Anthony, I'll merge this one and take your comment in a separate PR and send out this week.